src/InputManager.h

```diff
@@ -20,13 +20,17 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
 #pragma once
 
+#include <memory>
+
 #include <SDL2/SDL.h>
 
 #include "PlayerInput.h"
 #include "Vector.h"
 #include "BlobbyDebug.h"
 
-enum KeyAction
+#include "input_device/JoystickPool.h"
```

Comment:
I think here you can just forward declare `JoystickPool`, so we drag in less dependencies to this header.

Reply:
You are right. I change this.